### PR TITLE
Feature/plib 2511  example code pair component

### PR DESCRIPTION
**Pull request by kevinmpowell**
This builds on the html example list PR and adds some additional functionality to the code snippet component to support features required by the example code pair.

### CODE REVIEW

File: docs/.vuepress/components/cdr-doc-html-example-list.vue [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+      this.exampleCount = Object.keys(this.$slots).length;
+      for (const label in this.$slots) {
+        const mountId = '#cdr-doc-html-example-' + label;
+        const codeNode = this.extractCodeNodeFromVnodeTree(this.$slots[label][0]);
+        const templateSource = this.getStoredTemplateSourceForExample(label, codeNode);
+        this.renderExampleFromTemplate(templateSource, mountId);
+      }
+    },
+    methods: {
+      getStoredTemplateSourceForExample(label, codeNode) {
```
Comment (mhewson) [inline docs/.vuepress/components/cdr-doc-html-example-list.vue]: can you add comments to these methods

File: docs/.vuepress/components/cdr-doc-html-example-list.vue [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+          vNode.children.some(childNode => {
+            codeNode = this.extractCodeNodeFromVnodeTree(childNode);
+            return codeNode !== false;
+          });
+        }
+
+        return codeNode
+      },
+      renderExampleFromTemplate (template, mountId) {
+        // This method calls the Vue compiler directly to render the VUE template content as HTML
```
Comment (mhewson) [inline docs/.vuepress/components/cdr-doc-html-example-list.vue]: this is required to convert snippet from string back to vue?

File: docs/.vuepress/components/cdr-doc-example-code-pair.vue [hunk 3]
```diff
@@ -0,0 +1,292 @@
...
+  
+  .cdr-doc-item-background-toggle__button--active {
+    background: $partly-cloudy;
+  }
+
+  .cdr-doc-item-background-toggle__input {
+    position: absolute !important;
+    height: 1px; width: 1px; 
+    overflow: hidden;
+    clip: rect(1px 1px 1px 1px); /* IE6, IE7 */
```
Comment (mhewson) [inline docs/.vuepress/components/cdr-doc-example-code-pair.vue]: Not sure this is needed, these browsers have long been removed from the support matrix maintained by test at REI -

File: docs/.vuepress/components/cdr-doc-html-example-list.vue (hunk 1 above)
Comment (kevinmpowell) [reply docs/.vuepress/components/cdr-doc-html-example-list.vue]: yes

File: docs/.vuepress/components/cdr-doc-html-example-list.vue (hunk 2 above)
Comment (kevinmpowell) [reply docs/.vuepress/components/cdr-doc-html-example-list.vue]: That's correct

File: docs/.vuepress/components/cdr-doc-example-code-pair.vue (hunk 3 above)
Comment (kevinmpowell) [reply docs/.vuepress/components/cdr-doc-example-code-pair.vue]: I copy/pasted that code snippet from a google search on "accessibly hidden" text. I think those comments are stating that this is supported all the way back to IE6 and IE7, not that the `clip` property is used _to_ support IE6 and IE7. I'll remove the comment to reduce confusion.

Comment (kevinmpowell) [reply docs/.vuepress/components/cdr-doc-example-code-pair.vue]: Nope, I was wrong about this. I see that this was an old, alternate syntax. I've removed this line.

**Comment by kevinmpowell** [comment]
@mhewson I think you might have approved this earlier, but one of my changes invalidated it. Mind taking another look?

**Pull request merged**